addons/account/models/partner.py

```diff
@@ -86,15 +86,26 @@ def _validate_foreign_vat_country(self):
                             raise ValidationError(_("You cannot create a fiscal position with a foreign VAT within your fiscal country without assigning it a state."))
                         else:
                             raise ValidationError(_("You cannot create a fiscal position with a foreign VAT within your fiscal country."))
+                if record.country_group_id and record.country_id:
+                    if record.country_id not in record.country_group_id.country_ids:
+                        raise ValidationError(_("You cannot create a fiscal position with a country outside of the selected country group."))
 
                 similar_fpos_domain = [
                     ('foreign_vat', '!=', False),
-                    ('country_id', '=', record.country_id.id),
                     ('company_id', '=', record.company_id.id),
                     ('id', '!=', record.id),
                 ]
+
+                if record.country_group_id:
+                    foreign_vat_country_id = next(country_id for country_id in [record.country_id] + [country for country in self.country_group_id.country_ids] if country_id.code.upper() == self.foreign_vat[:2])
```

Comment:
If you call it foreign_vat_country_id, you're implying it's an id, so an integer. Here, it's a record; this causes confusion => foreign_vat_country